Libraries/xcdriver/Sources/UsageAction.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ Copyright (c) 2016-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <xcdriver/UsageAction.h>
+
+using xcdriver::UsageAction;
+using xcdriver::Options;
+using libutil::Filesystem;
+
+UsageAction::
+UsageAction()
+{
+}
+
+UsageAction::
+~UsageAction()
+{
+}
+
+int UsageAction::
+Run(Filesystem const *filesystem, Options const &options)
+{
+    // TODO: Perhaps have xcdriver/Tools/xcbuild.cpp pass the name of the
+    //       executable (`argv[0]`) to `xcdriver::Driver::Run()`? That way,
+    //       the `UsageAction` could display the name of the program, instead
+    //       of hardcoding "xcbuild".
```

Comment:
You could use this for the equivalent of `argv[0]`:

    FSUtil::GetBaseName(SysUtil::GetExecutablePath());

Both `FSUtil` and `SysUtil` are in the `libutil` library.